js/index.js

```diff
@@ -711,11 +420,13 @@ $(function() {
                     setupEditor();
                 }).fail(function() {
                     console.log("Unable to load gist contents");
+                    $('#code_textarea').val("-- Unable to load gist contents. Gist = " + id);
```

Comment:
Ok, I think we should try to break down `loadOptions` and/or `setupEditor` a bit, so that we don't end up looping, in that case.